pkg/reconciler/common/releases.go

```diff
@@ -74,42 +74,53 @@ func InstalledManifest(instance v1alpha1.KComponent) (mf.Manifest, error) {
 	return fetch(installedManifestPath(current, instance))
 }
 
-// IsUpDowngradeEligible returns the bool indicate whether the installed manifest is able to upgrade or downgrade to
-// the target manifest.
-func IsUpDowngradeEligible(instance v1alpha1.KComponent) bool {
+// IsVersionValidMigrationEligible returns the bool indicate whether the target version is valid and the installed
+// manifest is able to upgrade or downgrade to the target manifest.
+func IsVersionValidMigrationEligible(instance v1alpha1.KComponent) error {
+	var errMess error
```

Comment:
Since this is an actual error interface, and not a message, why not sticking to a traditional `err` variable name?

Reply:
there are multiple errors in this function.

Reply:
Sure, but they could all use the same variable from what I see. My point is that having both `errMss` and `err` within the same function doesn't really help understanding which one is used for what purpose (as opposed to something called `validationErr`, for example).